modules/oe_content_featured_media_field/src/Plugin/Field/FieldWidget/FeaturedMediaFieldWidget.php

```diff
@@ -0,0 +1,127 @@
+ * Plugin implementation of the 'oe_featured_media_widget' widget.
+ *
+ * @FieldWidget(
+ *   id = "oe_featured_media_widget",
+ *   label = @Translation("Featured media"),
+ *   description = @Translation("An autocomplete entity reference field and a text field."),
+ *   field_types = {
+ *     "oe_featured_media"
+ *   }
+ * )
+ */
+class FeaturedMediaFieldWidget extends EntityReferenceAutocompleteWidget {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $referenced_entities = $items->referencedEntities();
+
+    $element['featured_media'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Featured media'),
+    ];
+
+    // Append the match operation to the selection settings.
+    $selection_settings = $this->getFieldSetting('handler_settings') + [
+      'match_operator' => $this->getSetting('match_operator'),
+      'match_limit' => $this->getSetting('match_limit'),
+    ];
+
+    $element['featured_media']['target_id'] = [
+      '#type' => 'entity_autocomplete',
+      '#title' => $this->t('Media item'),
+      '#description' => $this->getMediaReferenceHelpText($selection_settings),
+      '#target_type' => $this->getFieldSetting('target_type'),
+      '#selection_handler' => $this->getFieldSetting('handler'),
+      '#selection_settings' => $selection_settings,
+      // Entity reference field items are handling validation themselves via
+      // the 'ValidReference' constraint.
+      '#validate_reference' => FALSE,
+      '#maxlength' => 1024,
+      '#default_value' => isset($referenced_entities[$delta]) ? $referenced_entities[$delta] : NULL,
+      '#size' => $this->getSetting('size'),
+      '#placeholder' => $this->getSetting('placeholder'),
+    ];
+    $element['featured_media']['caption'] = [
+      '#title' => $this->t('Caption'),
+      '#type' => 'textarea',
+      '#default_value' => $items[$delta]->caption,
+      '#rows' => 2,
+      '#weight' => $delta,
+      '#attributes' => ['class' => ['js-text-full', 'text-full']],
+    ];
+
+    return $element;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function massageFormValues(array $values, array $form, FormStateInterface $form_state) {
+    foreach ($values as $key => $value) {
+      $values[$key]['target_id'] = $value['featured_media']['target_id'];
+      $values[$key]['caption'] = $value['featured_media']['caption'];
+    }
+
+    return $values;
+  }
+
+  /**
+   * Gets extensive help text for media reference field.
+   *
+   * It provides useful information about the allowed media types to reference
+   * and a link to the media item list to create new media items.
+   *
+   * @param array $selection_settings
+   *   The entity reference field handler settings.
+   *
+   * @return mixed
+   *   The render array containing help texts.
+   */
+  protected function getMediaReferenceHelpText(array $selection_settings): array {
```

Comment:
This can be removed after changing the formElement.

Reply:
Without this the help text won't work. Event if we would use use the parent.